com.unity.shadergraph/Editor/ShaderFoundry/Utilities/VirtualTextureAttributes.cs

```diff
@@ -0,0 +1,127 @@
+using System;
+using System.Collections.Generic;
+
+namespace UnityEditor.ShaderFoundry
+{
+    internal class VirtualTextureAttribute
+    {
+        const string AttributeName = "VirtualTexture";
+        const string LayerCountParamName = "layerCount";
+
+        internal const int MaxLayerCount = 4;
+        internal int LayerCount = 2;
+
+        internal ShaderAttribute Build(ShaderContainer container)
+        {
+            var attributeBuilder = new ShaderAttribute.Builder(container, AttributeName);
+            attributeBuilder.Param(new ShaderAttributeParam.Builder(container, null, LayerCount.ToString()).Build());
+            return attributeBuilder.Build();
+        }
+
+        internal static VirtualTextureAttribute FindFirst(IEnumerable<ShaderAttribute> attributes)
+        {
+            var attribute = attributes.FindFirst(AttributeName);
+            if (attribute.IsValid)
```

Comment:
I had this question about another location, but just wanted to check that this FindFirst won't return null (or that it's ok to check IsValid on a null return).

Reply:
See previous comment, this is valid (struct vs. class)